code/modules/projectiles/guns/energy/laser.dm

```diff
@@ -23,50 +44,50 @@
 	desc = "An older model of the basic lasergun, no longer used by Nanotrasen's private security or military forces. Nevertheless, it is still quite deadly and easy to maintain, making it a favorite amongst pirates and other outlaws."
 	ammo_x_offset = 3
 
-
-/obj/item/weapon/gun/energy/laser/captain
-	name = "antique laser gun"
-	icon_state = "caplaser"
-	item_state = "caplaser"
-	desc = "This is an antique laser gun. All craftsmanship is of the highest quality. It is decorated with assistant leather and chrome. The object menaces with spikes of energy. On the item is an image of Space Station 13. The station is exploding."
+/obj/item/weapon/gun/energy/lasercannon
+	name = "laser cannon"
+	desc = "With the L.A.S.E.R. cannon, the lasing medium is enclosed in a tube lined with uranium-235 and subjected to high neutron flux in a nuclear reactor core. This incredible technology may help YOU achieve high excitation rates with small laser volumes!"
+	icon_state = "lasercannon"
+	item_state = "laser"
+	w_class = 4
 	force = 10
-	stamina_percentage = 0.6
-	origin_tech = null
-	var/charge_tick = 0
+	stamina_percentage = 0.4
+	flags =  CONDUCT
+	slot_flags = SLOT_BACK
+	origin_tech = "combat=4;materials=3;powerstorage=3"
+	ammo_type = list(/obj/item/ammo_casing/energy/laser/heavy)
 	ammo_x_offset = 3
 
-/obj/item/weapon/gun/energy/laser/captain/scattershot
-	name = "scatter shot laser rifle"
-	icon_state = "lasercannon"
+/obj/item/weapon/gun/energy/xray
+	name = "xray laser gun"
+	desc = "A high-power laser gun capable of expelling concentrated xray blasts that pass through multiple soft targets and heavier materials"
+	icon_state = "xray"
 	item_state = "laser"
-	desc = "An industrial-grade heavy-duty laser rifle with a modified laser lense to scatter its shot into multiple smaller lasers. The inner-core can self-charge for theorically infinite use."
-	origin_tech = "combat=5;materials=4;powerstorage=4"
-	ammo_type = list(/obj/item/ammo_casing/energy/laser, /obj/item/ammo_casing/energy/laser/scatter)
+	origin_tech = "combat=5;materials=3;magnets=2;syndicate=2"
+	ammo_type = list(/obj/item/ammo_casing/energy/xray)
+	pin = null
+	ammo_x_offset = 3
 
-/obj/item/weapon/gun/energy/laser/captain/scattershot/attack_self(mob/living/user as mob)
-	select_fire(user)
-	update_icon()
+/obj/item/weapon/gun/energy/laser/hypercannon
+	name = "laser hypercannon"
+	desc = "An advanced laser cannon that does more damage the farther away the target is."
+	icon_state = "laser_hypercannon"
+	item_state = "alc"
+	w_class = 4
+	force = 10
+	slot_flags = SLOT_BACK
+	fire_sound = 'sound/weapons/lasercannonfire.ogg'
+	origin_tech = "combat=4;materials=3;powerstorage=3"
+	cell_type = "/obj/item/weapon/stock_parts/cell/laser/hypercannon"
+	ammo_type = list(/obj/item/ammo_casing/energy/laser/hyper)
 
-/obj/item/weapon/gun/energy/laser/captain/New()
+/obj/item/projectile/beam/laser/accelerator/Range()
```

Comment:
well why is the beam code in laser gun dm?ya sure this is the right place for it?